### ISSUE TITLE
[REVIEW]: Multiscale Solar Water Heating

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2695]
**Submitting author:** @milicag (<a href="http://orcid.org/0000-0003-1640-3931">Milica Grahovac</a>)
**Repository:** <a href="https://github.com/LBNL-ETA/MSWH" target ="_blank">https://github.com/LBNL-ETA/MSWH</a>
**Version:** v2.0.0
**Editor:** @sjpfenninger
**Reviewer:** @brynpickering, @nmstreethran
**Archive:** <a href="https://doi.org/10.5281/zenodo.4305504" target="_blank">10.5281/zenodo.4305504</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331/status.svg)](https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331"><img src="https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331/status.svg)](https://joss.theoj.org/papers/6f1ca3f2573fb48a7941510fd735d331)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/LBNL-ETA/MSWH" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@brynpickering & @nmstreethran, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sjpfenninger know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/LBNL-ETA/MSWH">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@milicag) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/LBNL-ETA/MSWH">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@milicag) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @brynpickering, @nmstreethran it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2695 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2695 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2695 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2695 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by sjpfenninger** [comment]
@milicag Can you confirm which branch of the repository the review should be looking at? I see paper.md both in `master` and `issue5_submit_to_joss`. And can you move paper.md and associated files into a subdirectory inside the repository root.

**Comment by milicag** [comment]
@sjpfenninger Please review the master branch. I have a few clarity questions. Should the sub-directory be of a certain name? Is it correct to assume that the associated files are: `paper.md`, `paper.bib` and `LICENSE.md`?

**Comment by milicag** [comment]
@sjpfenninger I moved the  `paper.md`, `paper.bib` and `LICENSE.md` into a new directory called `paper`. Let me know if you need anything else.

**Comment by sjpfenninger** [comment]
@milicag Thanks - but `LICENSE.md` should always be in the root of your repository (this is independent of your JOSS submission)

**Comment by sjpfenninger** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by sjpfenninger** [comment]
@whedon generate pdf from branch master

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch master. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2695 with the following error: 

 sh: 1: cd: can't cd to tmp/2695
/app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2695 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-364ded062842/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by sjpfenninger** [comment]
@openjournals/dev Any idea what is going wrong here? I suspect it is connected to the fact that `https://github.com/LBNL-ETA/MSWH/tree/issue5_submit_to_joss` is set as the submission repository.

**Comment by milicag** [comment]
@sjpfenninger Thanks, I moved the license back to the root. Just as a note, the paper compiled fine when it was on the root of the master. I deleted that branch as the PR was merged.

**Comment by xuanxu** [comment]
> @openjournals/dev Any idea what is going wrong here? I suspect it is connected to the fact that `https://github.com/LBNL-ETA/MSWH/tree/issue5_submit_to_joss` is set as the submission repository.

Yeah, that was it, I've updated the URL and now it should work

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@xuanxu Great, thanks! @milicag Looks like it's resolved. Might not have been the subdir, but I suggest keeping it anyway to make things neater.

**Comment by sjpfenninger** [comment]
@brynpickering , @nmstreethran I see that you still have unchecked items on your checklists -- is this because you are waiting for @milicag to address them? Could you give a brief update on what is still outstanding so that we can move this forward?

**Comment by brynpickering** [comment]
@sjpfenninger, we ended up moving discussion to the software repo, but I can give an update from my side. Overall, there's a lot of content in the submission, but I have become unsure as to whether it fits within the scope of JOSS. This concern aside, I have no new issues to raise, and see my review of this submission to be close to completion. Below are more specific comments which are being / have been addressed in individual issues.

Software
---------
The code looks to be sound, and the tests are thorough, if not entirely comprehensive; triggering of several error/info messages is not being caught by any tests, but core mathematical functionality is well tested.  The implementation could be considered a bit impenetrable for would-be contributors, since variable names aren't that human-readable.

Documentation
---------------
There is API documentation, and a Jupyter notebook as an example of a model, but otherwise the documentation is quite sparse. It probably meets the minimum JOSS requirements, but makes no attempt to go into detail on what's going on in the internals, nor to more thoroughly document the vast array of input parameters expected from the user (most of which have default values). We've gone in circles about installation instructions, but seem to have reached a point now where everyone is happy.

Paper
------
Following my review (https://github.com/LBNL-ETA/MSWH/issues/31), the paper has recently undergone a major revision, which I will ask Whedon to re-generate following the merging of the changes. From a quick look, I expect to only make minor comments on this update to the manuscript. 

Current concern
----------------
The software is open-source and well written, but it seems to be an archive of a piece of software produced for a specific (past) project. The original focus of the paper (https://github.com/LBNL-ETA/MSWH/issues/31), and a separate conversation about contribution guidelines (https://github.com/LBNL-ETA/MSWH/issues/32), raises the question of whether there is any expectation that a community grows around developing this software further. The past project(s) focussed on California, and although it is technically feasible to model anywhere in the world, the existing parametrisation and data processing retains that geographic focus. From the PR for an update to the paper:

> The water consumption profiles can be highly location specific and their development for additional climate zones would require new research efforts, although a quick approximation may be made with caution by scaling the California profiles to match the location specific estimate of the average annual water use.

**I would appreciate it if this concern could be addressed by the editorial team**. Perhaps it shouldn't be a concern at all.

**Comment by milicag** [comment]
@sjpfenninger @sjpfenninger @nmstreethran I'm happy to address any outstanding issues! 

Regarding the current concern section, I added a sentence to [the current PR branch](https://github.com/LBNL-ETA/MSWH/pull/33) that makes the quoted out section of the paper sound more as I had intended - the section is referring to future research. I don't think that any developer of software can provide all the possible inputs and for that reason I do give a bit of a guidance in the paper. It is essentially a tip to the user who does not reside in California. For that reason there is also an example for Bosnia. The paper by JOSS approach should be really short. The journal is, what draw me towards it, dedicated to software itself. I believe we should refocus our efforts here on the software itself.

All variables that are user facing are described in the code documentation, in each method or class. The code would be very long and hard to understand from the coding point of view if I was to use full words for each variable. I hope you understand this.

The software package is complete and that is also why I decided it is ready to be published as such. Any future research depends on the available funding and I cannot see how this relates to current state and JOSS publication. I would certainly love to have future contributions!

Please feel free to provide comment and I am here to clarify anything you might still be wondering about.

Thanks,
Milica

**Comment by sjpfenninger** [comment]
@milicag @brynpickering Thanks both for the update. I see no major issue here. @milicag You could explicitly point to the California-specific data as examples included, but they do not preclude application of the software elsewhere. @nmstreethran Could you provide an update what is outstanding from your side?

**Comment by nmstreethran** [comment]
Hi @sjpfenninger, sorry for the late response.

Based on my checklist, I only have one item remaining, which is a statement of need in the documentation. I think just a summary of the target audience would suffice as the usage of the software has already been described in the documentation [here](https://lbnl-eta.github.io/MSWH/source/models.html). I have checked off all other items as I believe PRs https://github.com/LBNL-ETA/MSWH/pull/33 and https://github.com/LBNL-ETA/MSWH/pull/35 will resolve them.

I opened issues https://github.com/LBNL-ETA/MSWH/issues/13, https://github.com/LBNL-ETA/MSWH/issues/14, https://github.com/LBNL-ETA/MSWH/issues/15, and https://github.com/LBNL-ETA/MSWH/issues/26 as part of my review, which have all been resolved. I can confirm that the software installation, example notebooks, web application, and tests run as expected.

Overall, I think this is a useful piece of software and I am happy to recommend to accept this submission once the outstanding PRs have been merged and the documentation is complete.

**Comment by milicag** [comment]
@sjpfenninger At @brynpickering's suggestion we are having a native speaker taking a final look at the `paper.md` on Monday and with that the current pull request will provide all edits the reviewers identified as needed. 

I have put together [this finalization issue](https://github.com/LBNL-ETA/MSWH/issues/16) that I intend to address as the last step, given your approval.

**Comment by milicag** [comment]
@brynpickering and @nmstreethran I just merged into the main branch the two PRs in response to the latest reviewer comment. 

[This PR](https://github.com/LBNL-ETA/MSWH/pull/35) brought in the contribution guidelines and added a short statement of need to the README file. After Robert, one of the co-authors provided a review I introduced his suggested edits and merged into main through this [PR](https://github.com/LBNL-ETA/MSWH/pull/33). I compiled the paper using your preview capabilities.

[This issue](https://github.com/LBNL-ETA/MSWH/issues/16) summarizes the remaining steps I would like to take next.

Please let me know if there is anything additional that needs to be done.

**Comment by milicag** [comment]
> @milicag @brynpickering Thanks both for the update. I see no major issue here. @milicag You could explicitly point to the California-specific data as examples included, but they do not preclude application of the software elsewhere. @nmstreethran Could you provide an update what is outstanding from your side?

I believe this is well addressed in the revised paper, however, do let me know if you find that it needs additional explanation in the paper.

**Comment by brynpickering** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: :point_left:

**Comment by milicag** [comment]
@brynpickering @nmstreethran I will wait for your prompt to start addressing [the remaining issue](https://github.com/LBNL-ETA/MSWH/issues/16), as I'd like to tag a code release that goes with the JOSS paper, to avoid a need to re-tag.

**Comment by milicag** [comment]
Please don't hesitate to ask for edits if you find something is still needed in the paper, I am happy to address it.

**Comment by brynpickering** [comment]
@milicag thanks for your work on this, I'm happy with the result and have completed my review checklist. 

A minor point (which I think you brought up in a now-closed issue/PR) is that the section of your summary relating to the components of the model could be placed in a sub-section (e.g. 'model components'), to break up the summary slightly. Similarly, a sub-section for 'future applications'/'user groups' in the Statement of Need section would work well. Not sure if this violates some strict structure that JOSS papers must adhere to (@sjpfenninger?), though.

**Comment by milicag** [comment]
@brynpickering Thank you! This is good news for me and the team. The paper is pretty short, only four pages of text, so I don't see a real reason to break it into the subsections, though I also see how that would help if someone is to read the paper really quickly. I leave it to @sjpfenninger.

**Comment by milicag** [comment]
@brynpickering, @nmstreethran - Thank you for your thorough review, it has improved many aspects of the software, especially the accessibility to users and contributors! I appreciate all your help, suggestions, and all our discussions. This was a really unique experience in these pandemic times.

**Comment by nmstreethran** [comment]
Thanks @milicag, everything looks good to me. I have completed my checklist as well. Well done to you and your team!

**Comment by milicag** [comment]
Thank you @nmstreethran.

**Comment by milicag** [comment]
The version number: v2.0.0
DOI: 10.5281/zenodo.4285623

@sjpfenninger

**Comment by milicag** [comment]
Hi @sjpfenninger, please don't hesitate to point out anything you would prefer to have edited, as I can introduce the edits and retag the release.

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by sjpfenninger** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1080/19401493.2013.765506 is OK
- 10.1016/j.apenergy.2017.07.093 is OK
- 10.2172/1025650 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by sjpfenninger** [comment]
Thanks @nmstreethran and @brynpickering for your extensive reviews.

@milicag: We are nearing completion on this. However, your paper is on the long side. We ask for papers to be 250-1000 words long and the more extensive information should really live in the project's documentation:

I think this can be addressed with two easy changes:

1. In the Summary section, move everything between the paragraphs starting with "We developed component sizing rule" and "The MSWH software is both accessible" into the documentation, with the exception of the paragraph starting with "The MSWH software was used to perform".

2. Move the statement of need to the documentation (maybe merging with https://lbnl-eta.github.io/MSWH/source/models.html, as suggested by @brynpickering), keeping the first three paragraphs (up to the one starting with "Modelica buildings library") in the paper, and maybe adding a fourth paragraph that summarises anything important from the remainder of the removed section including future applications and possible users.

**Comment by milicag** [comment]
@sjpfenninger I moved the parts of the paper, as you suggested, into the documentation. More specifically, the portions of the statement of need now are hosted [here](https://lbnl-eta.github.io/MSWH/source/models.html#future-applications-statement-of-need), while the parts related to the approach and methodology can be found [here](https://lbnl-eta.github.io/MSWH/source/models.html#approach-to-component-and-system-modeling-and-simulation). The paper is now significantly shorter. I summarized the parts that were extracted from the paper where I thought it was suitable.

Please let me know if you need any further edits. I can retag once we have the final version.

**Comment by milicag** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by milicag** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.11578/dc.20190524.2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02695/joss.02695/10.21105.joss.02695.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@milicag I made some minor suggestions for edits in https://github.com/LBNL-ETA/MSWH/pull/39. One additional minor point: you refer to a "12-core processor" and run time on that system, can you specify what type of processor to make this a more useful benchmark?

**Comment by milicag** [comment]
@sjpfenninger Thanks for your edits. I merged the PR. I specified the processor by just copying the info from my system. Please let me know if there is anything else that needs to be done before the re-tag.

**Comment by sjpfenninger** [comment]
@milicag Great, thanks. So yes, we need you to make a tagged release on GitHub and archive that release, for example on Zenodo or figshare, then report the version number and archive DOI back here in the review issue. Make sure the archive has the same author list and title as the paper does.

**Comment by milicag** [comment]
@sjpfenninger, here are the DOI of the v2 archive at Zenodo and a new release version tag:

[The tagged release](https://github.com/LBNL-ETA/MSWH/releases/tag/v2.0.0) with a version number v2.0.0
[DOI: 10.5281/zenodo.4305504](https://zenodo.org/record/4305504)

**Comment by sjpfenninger** [comment]
Hm, the Zenodo deposition says "Closed Access -Files are not publicly accessible."

Not sure what went wrong there, but can you fix that?

**Comment by sjpfenninger** [comment]
@whedon set v2.0.0 as version

**Comment by whedon** [comment]
OK. v2.0.0 is the version.

**Comment by milicag** [comment]
@sjpfenninger Zenodo was asking me to associate the code archive with an another license, I thought it was much safer to provide a link to our repository as an archive that contains the zip files ready for download of the version 2.0.0 of the software including the paper. The users can use the links provided in the Zenodo archive description to download the release from the Github repository. Is there an issue with this approach?

**Comment by milicag** [comment]
@sjpfenninger This is the [initial link on the Zenodo](https://github.com/LBNL-ETA/MSWH/releases/tag/v2.0.0) summary. This way I preserve only one layer of licensing that was granted to us through DOE code publication.

**Comment by milicag** [comment]
@sjpfenninger I could change the link to be [a direct download of the release v2.0.0](https://github.com/LBNL-ETA/MSWH/archive/v2.0.0.zip), or potentially better I could add that link in addition to the above link.

**Comment by milicag** [comment]
With my approach I avoided a need to use an additional license, while obtaining a DOI for the archive and providing full access to the code under a JOSS accepted license.

**Comment by milicag** [comment]
@brynpickering Just to avoid any potential confusion, my knowledge in the domain of license interaction, especially about aspects such as which license takes precedence and where and how that is exactly defined, and whether it might change in the future, is limited. Therefore I found a simplified solution that I don't see imposes any limitations provided the information available at the JOSS website. I am certainly open to hearing your view, in particular if there is something within the JOSS rules that would conflict with the approach I've taken. Please do not hesitate to ask any questions you might have, I am happy to learn more.

**Comment by sjpfenninger** [comment]
@milicag We are not asking you to chose a different license for the Zenodo archive. I do see that the LBNL BSD license is not a pre-defined choice on Zenodo. However you can choose as license "Other (Open)", then ensure that your license file which contains the BSD LBNL license is included at the root level of the Zenodo deposition (this should be no issue, as it's in the root of your repository).

We do need the archive on Zenodo to be a full archive of the submitted code, not just an empty submission to get a DOI. Let me know if this clears things up!

**Comment by milicag** [comment]
@sjpfenninger Sounds good. I just modified the archive.

**Comment by sjpfenninger** [comment]
@milicag: Great. The license is set to "Creative Commons Attribution 4.0 International" though. Can you set it to "Other (Open)"?

**Comment by sjpfenninger** [comment]
@milicag Actually I checked with Zenodo support and the license should be available However it is called "Lawrence Berkeley National Labs BSD variant license" on Zenodo (if you enter some or all of those terms in the search field on upload, you should see it). If you can pick that license, the Zenodo deposition gets the correct license metadata and all should be well.

**Comment by milicag** [comment]
@brynpickering That makes more sense, it shows in the search box after I start typing. The archive license should now be set to LBNLBSD. Thank you for checking with them.

**Comment by sjpfenninger** [comment]
@whedon set 10.5281/zenodo.4305504 as doi

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by sjpfenninger** [comment]
@whedon set 10.5281/zenodo.4305504 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4305504" target="_blank">10.5281/zenodo.4305504</a> is the archive.

**Comment by sjpfenninger** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by sjpfenninger** [comment]
Thanks @milicag for bearing with us. Looks all good now!

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1968

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1968, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.11578/dc.20190524.2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by milicag** [comment]
@sjpfenninger, this was an interesting and new journey! Thank you all for the contibutions and improvements to the documentation, the code, and the paper. I learned a lot.

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1338461089626017792) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1983
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02695](https://doi.org/10.21105/joss.02695)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@brynpickering, @nmstreethran - many thanks for your reviews here and to @sjpfenninger for editing this submission. JOSS relies upon the volunteer efforts of people like yourselves, we simply wouldn't be able to do it without you! :sparkles:

@milicag - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02695/status.svg)](https://doi.org/10.21105/joss.02695)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02695">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02695/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02695/status.svg
   :target: https://doi.org/10.21105/joss.02695
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02695/status.svg)](https://doi.org/10.21105/joss.02695)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
